packages/vite/src/node/plugins/html.ts

```diff
@@ -172,15 +172,32 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
     isDataUrl(url) ||
     checkPublicFile(url, config)
 
+  const inputFiles = new Set<string>()
+
   return {
     name: 'vite:build-html',
 
-    buildStart() {
+    buildStart({ input }) {
       isAsyncScriptMap.set(config, new Map())
+
+      let allInputs: string[]
+      if (typeof input === 'string') {
+        allInputs = [input]
+      } else if (Array.isArray(input)) {
+        allInputs = input
+      } else {
+        allInputs = Object.values(input)
+      }
+
+      for (const filename of allInputs) {
+        if (filename.endsWith('.html')) {
+          inputFiles.add(filename)
+        }
+      }
     },
 
     async transform(html, id) {
-      if (id.endsWith('.html')) {
+      if (inputFiles.has(id)) {
```

Comment:
so. in case there's already a plugin, it won't be added to the "bundler inputs"?

Reply:
Not sure what you mean. Is it not safe to use the list of inputs in `buildStart` as the list of inputs?